analysis/src/python/evaluation/benchmark/benchmark.py

```diff
@@ -0,0 +1,190 @@
+from analysis.src.python.evaluation.hyperstyle.evaluation_config import HYPERSTYLE_TOOL_PATH, HyperstyleEvaluationConfig
+from analysis.src.python.evaluation.qodana.evaluate import parse_qodana_result
+from analysis.src.python.evaluation.qodana.evaluation_config import QodanaEvaluationConfig
+from analysis.src.python.evaluation.utils.pandas_utils import get_language_version
+from analysis.src.python.evaluation.utils.solutions_saving_utils import save_solutions_to_files
+from analysis.src.python.utils.df_utils import read_df, write_df
+from analysis.src.python.utils.file_utils import create_directory
+
+logger = logging.getLogger(__name__)
+
+
+class Analyzer(Enum):
+    HYPERSTYLE = 'hyperstyle'
+    QODANA = 'qodana'
+
+    @classmethod
+    def from_value(cls, value: str) -> Optional['Analyzer']:
+        try:
+            return Analyzer(value)
+        except ValueError:
+            return None
+
+    @classmethod
+    def available_values(cls) -> Set[str]:
+        return {cls.HYPERSTYLE.value, cls.QODANA.value}
+
+
+def measure_run_time(f: Callable[[], Any], repeat: int, output_path: Path, parser: Callable[[Path], Any]) -> float:
+    repeat_times = []
+    for i in range(1, repeat + 1):
+        logger.info(f'Time measuring attempt={i}')
+
+        start_time = time.time()
+        f()
+        end_time = time.time()
+
+        difference = end_time - start_time
+        logger.info(f'Time: {difference}')
+        logger.info(parser(output_path))
+
+        repeat_times.append(difference)
+
+    return np.array(repeat_times).mean()
+
+
+def time_benchmark_row(
+    row: pd.Series,
+    docker_path: str,
+    analyzer: Optional[Analyzer],
+    repeat: int,
+    n_cpu: Optional[int],
+    tmp_directory: Path,
+) -> float:
+    if analyzer == Analyzer.HYPERSTYLE:
+        config = HyperstyleEvaluationConfig(
+            docker_path=docker_path,
+            tool_path=HYPERSTYLE_TOOL_PATH,
+            allow_duplicates=False,
+            with_all_categories=False,
+            new_format=False,
+            n_cpu=n_cpu,
+            tmp_path=tmp_directory,
+        )
+        parser = parse_hyperstyle_result
+    elif analyzer == Analyzer.QODANA:
+        config = QodanaEvaluationConfig(tmp_path=tmp_directory)
+        parser = parse_qodana_result
+    else:
+        raise NotImplementedError(f'Benchmark for {analyzer} is not implemented.')
+
+    logger.info(f'Benchmarking {row[SubmissionColumns.ID.value]} ...')
+
+    language_version = get_language_version(row[SubmissionColumns.LANG.value])
+    input_path = create_directory(config.tmp_path / 'input', clear=True)
+    output_path = create_directory(config.tmp_path / 'output', clear=True)
+
+    save_solutions_to_files(row.to_frame().T, language_version, input_path, config.with_template)
+
+    command = config.build_command(input_path, output_path, language_version)
+
+    mean_time = measure_run_time(
```

Comment:
probably it is better to add a cli argument to choose what time to check the user wants: mean or median. What do you think?